shotover-proxy/src/protocols/cassandra_protocol2.rs

```diff
@@ -396,7 +400,7 @@ impl CassandraCodec2 {
         let mut projection: Vec<String> = Vec::new();
         let mut primary_key: HashMap<String, Value> = HashMap::new();
         let mut ast: Option<Statement> = None;
-        let parsed_sql = Parser::parse_sql(&dialect, query_string);
+        let parsed_sql = Parser::parse_sql(&dialect, &query_string);
```

Comment:
now that we dont need ownership of the string we should just take an `&str`. (This will save an allocation for the caller)